portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -48,4 +61,21 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.setContentType("application/json;");
     response.getWriter().println(gson.toJson(comments));
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String text = request.getParameter("text-input");
+    String author = request.getParameter("author");
+    Date date = new Date();
+
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("text", text);
+    commentEntity.setProperty("author", author);
+    commentEntity.setProperty("date", date);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
+
+    response.sendRedirect("/#comments");
```

Comment:
Please add a comment to explain why you would like to do this.

Reply:
Done! I added a few comments to both doGet() and doPost()